### PR TITLE
Add contributing file

### PR DESCRIPTION
**Pull request by shonfeder**
- Adds a very rough invitation to contribute (to be expanded later)
- Adds a DCO requirement with instructions

I've also submitted a request to add the DCO app, which @greg-szabo approved. Thanks to @romac for pointing it out.

### CODE REVIEW

**Comment by shonfeder** [comment]
I initially opened this PR without a signed commit, and the DCO app failed it, the I reworded the commit to include it and we're passing :+1:

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+There are two principles ways to contribute at this point:
+
+- [File an issue][] with a bug report or feature request.
+- Contribute to the source code (including documentation).
+
+## Contributing to the source
+
+We require that all commits be signed, certifying the [Developer Certificate
+of Origin (DCO)][DCO]. By singing your commits you confirm that you wrote
```
Comment (thanethomson) [inline CONTRIBUTING.md:14]:
```suggestion
of Origin (DCO)][DCO]. By signing your commits you confirm that you wrote
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+your contribution, or have otherwise secured the right to contribute the
+contribution as an open-source addition to our project.
+
+Here is the full text, also hostd at https://developercertificate.org/ :
```
Comment (thanethomson) [inline CONTRIBUTING.md:18]:
```suggestion
Here is the full text, also hosted at https://developercertificate.org/ :
```

**Comment by shonfeder** [comment]
Fixed (after some git foundering :). Thanks for the corrections!

**Pull request merged**